lib/did-io.js

```diff
@@ -187,6 +187,11 @@ api.generateHash = function(email, passphrase) {
  * @return true if the DID's format is valid, false if not.
  */
 api.validateDidFormat = function(did) {
+  // TODO: implement validation of specific methods
+  // for now, any did:foo:<string> is valid
+  if((did.match(/\:/g) || []).length === 2) {
+    return true;
+  }
   var didRegex = new RegExp('^did\:[0-9a-f]{8}-[0-9a-f]{4}-' +
     '4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$', 'i');
   return didRegex.test(did);
```

Comment:
I'd suggest pulling out the static regex setup and using both an old uuid style and new spec method style regex.  Just check the start part of the did for the new style.  Some lightly tested code:
```js
// old did:uuidv4 style
const didUuidRegex = new RegExp('^did\:[0-9a-f]{8}-[0-9a-f]{4}-' +
￼      '4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$', 'i');
// start of generic did scheme before path and fragment
// FIXME: accepts bad trailing ":" before path: "did:method:specific:idstring:/path
const didGenericRegex = /^did:([a-z0-9]+):([A-Za-z0-9.-]+(?::[A-Za-z0-9.-]+)*)/
api.validateDidFormat = function(did) {
  return didUuidRegex.test(did) || didGenericRegex.test(did);
￼}
￼```